File: sgi-webapp/src/app/core/models/csp/relacion-ejecucion-economica.ts
import { marker } from "@biesbjerg/ngx-translate-extract-marker";
import { I18nFieldValue } from "@core/i18n/i18n-field";
import { DateTime } from "luxon";
import { IProyectoSge } from "../sge/proyecto-sge";

export interface IRelacionEjecucionEconomica {
  id: number;
  nombre: I18nFieldValue[];
  codigoExterno?: string;
  codigoInterno?: string;
  fechaInicio: DateTime;
  fechaFin: DateTime;
  proyectoSge: IProyectoSge;
  tipoEntidad: TipoEntidad;
  fechaFinDefinitiva: DateTime;
}

export enum TipoEntidad {
  GRUPO = 'GRUPO',
  PROYECTO = 'PROYECTO'
}

export const TIPO_ENTIDAD_MAP: Map<TipoEntidad, string> = new Map([
  [TipoEntidad.GRUPO, marker('csp.relacion-ejecucion-economica.tipo-entidad.GRUPO')],
  [TipoEntidad.PROYECTO, marker('csp.relacion-ejecucion-economica.tipo-entidad.PROYECTO')]
]);

File: sgi-csp-service/src/main/java/org/crue/hercules/sgi/csp/dto/RelacionEjecucionEconomica.java
package org.crue.hercules.sgi.csp.dto;

import java.time.Instant;
import java.util.Collection;

import org.crue.hercules.sgi.framework.i18n.I18nFieldValueDto;

import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

@Data
@NoArgsConstructor
@AllArgsConstructor
@Builder
public class RelacionEjecucionEconomica {

  public enum TipoEntidad {
    /** Grupo */
    GRUPO,
    /** Proyecto */
    PROYECTO
  }

  private Long id;
  private Collection<I18nFieldValueDto> nombre;
  private String codigoExterno;
  private String codigoInterno;
  private Instant fechaInicio;
  private Instant fechaFin;
  private String proyectoSgeRef;
  private TipoEntidad tipoEntidad;
  private Instant fechaFinDefinitiva;

}
